ui/widgets/nuxeo-html-editor.js

```diff
@@ -55,121 +51,90 @@ import './alloy/alloy-ocean.js';
+        /**
+         * When set the content will be read only.
+         */
+        readOnly: {
+          type: Boolean,
+          value: false,
+          reflectToAttribute: true,
+        },
 
-        hideToolbars: Boolean,
+        _editor: {
+          type: Object,
+        },
       };
     }
 
-    static get importMeta() {
-      return import.meta;
+    static get observers() {
+      return ['_valueChanged(value, _editor)', '_readOnlyChanged(readOnly, _editor)'];
     }
 
-    connectedCallback() {
-      super.connectedCallback();
-      AlloyEditor._langResourceRequested = true; // skip loading of language file
-      const alloyI18Nscript = document.createElement('script');
-      alloyI18Nscript.async = true;
-      alloyI18Nscript.src = this.resolveUrl(`alloy/lang/alloy-editor/${CKEDITOR.config.language}.js`);
-      document.head.appendChild(alloyI18Nscript);
-      alloyI18Nscript.onload = () => {
-        setTimeout(() => {
-          this._init();
-        }, 100);
-      };
+    static get importMeta() {
+      return import.meta;
     }
 
-    _init() {
+    ready() {
+      super.ready();
       // init editor
-      const toolbars = this.hideToolbars
-        ? {}
-        : {
-            add: {
-              buttons: ['image', 'camera', 'hline', 'table'],
-              tabIndex: 2,
-            },
-            styles: {
-              selections: [
-                {
-                  name: 'link',
-                  buttons: ['linkEdit'],
-                  test: AlloyEditor.SelectionTest.link,
-                },
-                {
-                  name: 'image',
-                  buttons: ['imageLeft', 'imageCenter', 'imageRight'],
-                  test: AlloyEditor.SelectionTest.image,
-                },
-                {
-                  name: 'text',
-                  buttons: [
-                    'styles',
-                    'bold',
-                    'italic',
-                    'underline',
-                    'strike',
-                    'paragraphLeft',
-                    'paragraphCenter',
-                    'paragraphJustify',
-                    'ul',
-                    'ol',
-                    'quote',
-                    'link',
-                    'removeFormat',
-                  ],
-                  test: AlloyEditor.SelectionTest.text,
-                },
-                {
-                  name: 'table',
-                  buttons: ['tableRow', 'tableColumn', 'tableCell', 'tableRemove'],
-                  getArrowBoxClasses: AlloyEditor.SelectionGetArrowBoxClasses.table,
-                  setPosition: AlloyEditor.SelectionSetPosition.table,
-                  test: AlloyEditor.SelectionTest.table,
-                },
-              ],
-              tabIndex: 1,
-            },
-          };
-      const editor = AlloyEditor.editable(this.$.editor, {
-        readOnly: this.readOnly,
-        uiNode: this.root,
-        toolbars,
+      const toolbar = [
```

Comment:
Not sure if we should test the toolbar, since it is "our" definition and it seems that we have specific requirements (e.g. removing the formatting button?)

Reply:
"remove formatting button" is a button to remove formatting :P

I agree it would be nice to test the toolbar and perhaps alaso make it configurable but for now plan is to land a pretty much ootb Quill to validate it serves our purposes.

Reply:
Fair enough. So, let's create a ticket for that 😉